src/main/kotlin/net/irisshaders/lilybot/extensions/util/Tags.kt

```diff
@@ -0,0 +1,193 @@
+
+class Tags : Extension() {
+	override val name = "tags"
+
+	override suspend fun setup() {
+
+		publicSlashCommand(::TagArgs) {
+			name = "tag"
+			description = "The parent command for all /tag commands"
+
+			check { anyGuild() }
+
+			action {
+
+				if (DatabaseHelper.getTag(guild!!.id, arguments.tagName, "name") == null) {
+					respondEphemeral {
+						content = "Unable to find tag! Does this tag exist?"
+					}
+					return@action
+				}
+
+				respond {
+					embed {
+						color = DISCORD_BLURPLE
+						title = DatabaseHelper.getTag(guild!!.id, arguments.tagName, "tagTitle")!!.toString()
+						description = DatabaseHelper.getTag(guild!!.id, arguments.tagName, "tagValue")!!.toString()
+						timestamp = Clock.System.now()
+					}
+				}
+			}
+		}
+
+		publicSlashCommand {
+			name = "tag-help"
+			description = "Explains how the tag command works!"
+
+			action {
+				respond {
+					embed {
+						title = "How does the /tag command and others work?"
+						description =
+							"The tag command allows users to add guild specific 'tag' commands at runtime to their " +
+									"guild. Tags are like custom commands, they can do say what ever you want them" +
+									"to say.\n\nTo create a tag, if you have the Moderate Members permission, run " +
+									"the following command:\n`/create-tag <name> <title> <value>`\nYou will be " +
+									"prompted to enter a name for the tag, a title for the tag, and the value for the" +
+									" tag. This is what will appear in the embed of your tag. You can enter any " +
+									"character you like into all of these inputs.\n\nTo use the tag, run the " +
+									"following command\n`/tag <name>`\nYou will be prompted to enter the tag name, " +
+									"but will have an autocomplete window to aid you. The window will list all the " +
+									"tags that the guild has.\n\nTo delete a tag, if you have the Moderate Members " +
+									"permission, run the following command\n`/delete-tag <name>`\nYou will be " +
+									"prompted to enter the name of the tag, again aided by autocomplete, and then " +
+									"the tag will be completed.\n\nGuilds can have any number of tags they like. The " +
+									"limit on `tagValue` for tags is 1024 characters, which is the embed description " +
+									"limit enforced by discord"
+						color = DISCORD_BLURPLE
+						timestamp = Clock.System.now()
+					}
+				}
+			}
+		}
+
+		ephemeralSlashCommand(::CreateTagArgs) {
+			name = "create-tag"
+			description = "Create a tag for your guild!"
+
+			check { anyGuild() }
+			check { hasPermission(Permission.ModerateMembers) }
+
+			action {
+				val actionLogId = getConfigPrivateResponse("modActionLog") ?: return@action
+				val actionLog = guild!!.getChannel(actionLogId) as GuildMessageChannelBehavior
+
+				actionLog.createEmbed {
+					title = "Tag created!"
+					description = "${user.asUser().mention} created the tag ${arguments.tagName}"
```

Comment:
This is inconsistent with other logging. For anything that's moderator only, we usually only have a footer. Tag title can probably be moved to the title.